paper.md

```diff
@@ -2,11 +2,88 @@
 
+<!-- there's a problem of reproducibility in science -->
+
+Scientists and researchers often leave experimental artifacts
+like code, datasets, configuration files, etc. on open-access repositories like 
+Zenodo [@_zenodo_], Figshare [@_figshare_] or Github[@github] without proper
+instructions to reproduce after completing with their research. Therefore,
+it results in one wasting a lot of time trying to figure out how to
+reproduce the experiment from the archived artifacts. Hence, making the process 
+extremely inefficient and cumbersome. This problem in known as the Reproducibility 
+crisis or Replication crisis in science [@sep-scientific-reproducibility].
+
+![An end-to-end example of a workflow. On the left we have the 
+`.yml` file that defines the workflow. On the right, a pictorial 
+representation of it.](./figures/casestudy.png){#fig:casestudy}
+
+<!-- this is what people have tried with workflow engines -->
+
+The reproducibilty problem [@rosendaal2017reproducibility] has been attempted several times in the past which led to the development of various tools and frameworks which tried 
+to solve the problem in distinct ways [@peng2011reproducible]. Scientific workflow execution engines has 
+been a predominant solution for handling this problem by organizing the steps in a 
+complex scientific workflow as the nodes of a directed acyclic graph (DAG) and 
+executing them in correct order [@cohen2017scientific][@sciprocauto2009workflow][@albrecht2012makeflow].
+Ideally, workflows should improve the reproducibility of scientific applications 
+by making it easier to share and reuse between scientists. However, scientists 
+often find it difficult to reuse others workflows, which is known as _workflow decay_. [@workflow_decay]
+This is mostly because of the differences in the environment where the workflows are developed and the environment where they are being reproduced [@meng2017facilitating]. Some people have tried to solve this problem using the traditional Virtual machine model where they execute every task inside a separate VM [@gruning2018practical][@howe2012virtual]. People built virtual machine images from their development environment and used for setting up VM's in the Cloud. [@virtandnestedvirt2012]
+
+<!-- this is why it is still hard using VM's -->
+
+Although, the above approach solves the platform dependency problem very well, spawing 
+VM's to do granular steps of a workflow is very inefficient [@barik2016performance]. When several VM's are running on the same host, performance may be hindered if the computer it’s running on lacks sufficient power. This makes running workflows on commodity hardware extremely unlikely and inefficient. Resource utilization is very high in case of VM's 
+since every VM requires a copy of the operating system running on top of
+the host's operating system. In fact, a VM runs in a non-priviledged mode
+which does not have the capability to execute many privileged instructions. 
```

Comment:
this is irrelevant for what is being discussed (privileged vs non-privileged executions, etc).